public/lessonBuilder/lessonItemsController.js

```diff
@@ -97,28 +62,95 @@ angular.module('newApp')
         }
     }
 
+    function findTeacherRes(modules, moduleName) {
+        for(i = 0; i < modules.length; i++) {
+            // find the module tha has moduleName AND "Teacher" in it.
+            if(modules[i].name.search(moduleName) != -1) {
```

Comment:
It's recommended to use strict equality (`!==` or `===`) in pretty much every situation.  There is no difference here, but it's just a good habit to get into.

The loose version of equality testing does type casing before the test, such that `-1 == '-1'` is `true`.